assets/js/base/context/hooks/use-checkout-address.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import {
+	defaultAddressFields,
+	AddressFields,
+	EnteredAddress,
+	ShippingAddress,
+	BillingAddress,
+} from '@woocommerce/settings';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	useShippingDataContext,
+	useCheckoutContext,
+} from '../providers/cart-checkout';
+import { useCustomerData } from './use-customer-data';
+
+interface CheckoutAddress {
+	shippingAddress: ShippingAddress;
+	billingData: BillingAddress;
+	setShippingAddress: ( data: Partial< EnteredAddress > ) => void;
+	setBillingData: ( data: Partial< EnteredAddress > ) => void;
+	setEmail: ( value: string ) => void;
+	setPhone: ( value: string ) => void;
+	setShippingPhone: ( value: string ) => void;
+	shippingAsBilling: boolean;
```

Comment:
@ralucaStan The longer form would be "Use shipping address as billing address", or useShippingAsBilling. Is the short form too confusing?

Reply:
It is named like this because on the form you enter your shipping address first, and the checkbox controls whether this shipping address should be used as the billing address.

Reply:
> @ralucaStan The longer form would be "Use shipping address as billing address", or useShippingAsBilling. Is the short form too confusing?

@mikejolley For me it was confusing. Initially I thought it is inherited from the shortcode block, where the billing is added first.  With `useShippingAsBilling` it makes more sense.